bin/mympingpong.py

```diff
@@ -302,7 +302,12 @@ def run(self, abort_check=True, seed=1, msgsize=1024, maxruntime=0):
             'ppgroup': group,
         })
 
-        self.writehdf5(data, attrs, failed, fail)
+        if parallel_io or self.rank == 0:
+            self.writehdf5(data, attrs, failed, fail, parallel_io=parallel_io)
+        else:
+            self.log.debug("sending data to master rank...")
```

Comment:
add that this is blocking till the master rank listens for this rank (in loop below).